app/models/game.rb

```diff
@@ -102,6 +102,17 @@ def reviews_count
   end
 
   def state
-    aasm_state.tr('_', ' ').titleize
+    self.class.readable_state(aasm_state)
+  end
+
+  def self.states
```

Comment:
Can you comment the logic from line 108 onwards? It is a bit confusing to understand what you are doing on first read-through